### PR TITLE
Use a gRPC server to determine if a plugin should be multiplexed

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: sdk/database/dbplugin/v5/grpc_database_plugin.go [hunk 1]
```diff
@@ -44,6 +45,12 @@ func (d GRPCDatabasePlugin) GRPCServer(_ *plugin.GRPCBroker, s *grpc.Server) err
 			factoryFunc: d.FactoryFunc,
 			instances:   make(map[string]Database),
 		}
+
+		// Multiplexing is enabled for this plugin, register the server so we
+		// can tell the client in Vault.
+		pluginutil.RegisterPluginMultiplexingServer(s, pluginutil.PluginMultiplexingServerImpl{
+			Supported: true,
```
Comment (calvn) [inline sdk/database/dbplugin/v5/grpc_database_plugin.go:52]: Do you think it'd be useful to also register a `PluginMultiplexingServer` with `Supported` set to false on the case above to be able to differentiate between `Unimplemented` and an explicit `false`? That we we can detect the difference between a plugin that can register such server (i.e. an older plugin) vs one that choses not to.

**Review by calvn — APPROVED** [review]
Very neat suggestion! Only had one question around this.

File: vault/plugin_catalog.go [hunk 2]
```diff
@@ -234,7 +234,7 @@ func (c *PluginCatalog) newPluginClient(ctx context.Context, pluginRunner *plugi
 		id:     id,
 		logger: c.logger.Named(pluginRunner.Name),
 		cleanupFunc: func() error {
-			return c.removePluginClient(pluginRunner.Name, id)
```
Comment (briankassouf) [inline vault/plugin_catalog.go:237]: The context here is likely going to be expired by the time this cleanup function runs

Comment (calvn) [reply vault/plugin_catalog.go:237]: That's a good point. I was gonna suggest making `cleanupFunc` context-aware but `Close` doesn't accept a context so it's of no use. Would a background context with some safe timeout value be okay here?

Comment (fairclothjm) [reply vault/plugin_catalog.go:237]: If I update `pluginClient` to also store the multiplexing flag then we shouldn't need to pass the context as mentioned [here](https://github.com/hashicorp/vault/pull/14033/files#r806344620).

**Pull request merged**